src/server/game/Quests/QuestDef.cpp

```diff
@@ -252,15 +252,19 @@ int32 Quest::GetRewOrReqMoney(uint8 playerLevel) const
         }
     }
 
-    return static_cast<int32>(rewardedMoney * sWorld->getRate(RATE_REWARD_BONUS_MONEY));
+    return static_cast<int32>(rewardedMoney * sWorld->getRate(RATE_REWARD_QUEST_MONEY));
 }
 
 uint32 Quest::GetRewMoneyMaxLevel() const
 {
+    uint32 rewMoney = 0;
+
     if (HasFlag(QUEST_FLAGS_NO_MONEY_FROM_XP))
-        return 0;
+        return rewMoney;
+
+    rewMoney = (XPValue(sWorld->getIntConfig(CONFIG_MAX_PLAYER_LEVEL)) * (6 * COPPER));
 
-    return static_cast<int32>(RewardBonusMoney * sWorld->getRate(RATE_REWARD_BONUS_MONEY));
+    return static_cast<int32>(rewMoney * sWorld->getRate(RATE_REWARD_BONUS_MONEY));
```

Comment:
I think this doesn't make sense. `RewardBonusMoney * ConfigRate` should be correct as the Bonus Money comes from quest_template.RewardBonusMoney (RewardBonusMoney var). By changing this you render this DB field useless

Reply:
So either it is a fixed rate and then RewardBonusMoney field should be deleted.
Or there are some cases where a quest has a different bonus reward, to which changes need to be made to take RewardBonusMoney into account.

Also I would like to know where the 6 comes from

Reply:
> By changing this you render this DB field useless

Yes. The RwardBonusMoney field for quest_template entries not only change every time max level moves on Blizz servers, but for some quests (such as Klaven's Tower as mentioned in the PR description), it changes with your player level. It doesn't matter since afaik the reward value doesn't get used until you are at max level, but the field itself isn't fixed on Blizz. I can add a hard-coded check for if the max player level config is 80, and only *then* use the RewardBonusMoney value, but that still wouldn't account for dynamic reward values, plus hard-coding sucks just to use the same value as the calculation yields. Plus, according to [an old version of the XP Wowpedia article](https://wowpedia.fandom.com/wiki/Experience_point?oldid=2148651), depending on the expansion you owned and what your own max level was, the bonus would be different.

Trust me, it hurts to make a brute-forced value useless, but in this case it's wrong to keep using it.

> Also I would like to know where the 6 comes from

The formula for the reward has been known for [a long time](https://wowpedia.fandom.com/wiki/Quest?oldid=1035002), and it has been correct for every single quest I've looked at. Apparently it changed post-Wrath, but luckily we're not dealing with that.

Reply:
1. Okay then we delete that DB column. Otherwise, people will think it still is used. You will also need to delete the reference in the conf.
2. Aight. Can you add a comment with the link to wowpedia so nobody goes and changes it in the future

Reply:
> Okay then we delete that DB column. [...] You will also need to delete the reference in the conf. [...] Can you add a comment with the link to wowpedia [...]

Done.